src/csvcubed/models/sparqlresults.py

```diff
@@ -393,11 +395,66 @@ def _map_qube_component_sparql_result(
     return result
 
 
+def _get_column_components_and_check_for_cube_shape(
+    component: QubeComponentResult,
+    cube_shape: CubeShape,
+    csv_column_definitions: List[ColumnDefinition],
+    observed_value_columns: List[ColumnDefinition],
+):
+
+    if cube_shape == CubeShape.Standard and component.property_type == "Measure":
+        measure_column = first(
+            csv_column_definitions,
+            lambda c: c.property_url == "http://purl.org/linked-data/cube#measureType",
+        )
+
+        if measure_column is None:
+            raise KeyError("Could not find standard shape measure column.")
+
+        component.real_columns_used_in = [measure_column]
+
+        obs_val_column = first(
+            csv_column_definitions,
+            lambda c: c.property_url is not None
+            and c.data_type is not None
+            and measure_column is not None  # Here to satisfy buggy pyright.
+            and measure_column.name in uritemplate.variables(c.property_url),
```

Comment:
Can you put a comment above here stating that we *assume* that the only column which is permitted to reference the measure column is the observed values column.